geoportailv3/static/js/layerinfo/layerinfoservice.js

```diff
@@ -14,32 +14,112 @@ goog.require('ngeo.CreatePopup');
 app.ShowLayerinfo;
 
 
+
 /**
  * @param {angular.$http} $http Angular $http service
  * @param {angular.$sce} $sce Angular $sce service
+ * @param {angular.Scope} $rootScope The rootScope provider.
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
  * @param {ngeo.CreatePopup} ngeoCreatePopup Ngeo popup factory service
- * @return {app.ShowLayerinfo} The show layer info function.
+ * @constructor
  * @ngInject
  */
-app.showLayerinfoFactory = function($http, $sce, ngeoCreatePopup) {
+app.showLayerinfoFactory = function($http, $sce, $rootScope,
+    gettextCatalog, ngeoCreatePopup) {
 
   /**
    * @type {ngeo.Popup}
    */
   var popup = ngeoCreatePopup();
+  /**
+   * @type {Object.<string, !angular.$q.Promise>}
+   * @private
+   */
+  var promises_ = {};
+
+  /**
+   * @type {ol.layer.Layer}
+   */
+  var currentLayer = null;
+
+  $rootScope.$on('gettextLanguageChanged', function() {
+    if (!goog.isNull(currentLayer) && popup.getOpen()) {
+      showLayerInfo(currentLayer);
+    }
+  });
+
+  return showLayerInfo;
+
+  /**
+   * @param {ol.layer.Layer} layer The layer
+   * @this {Object}
+   */
+  function showLayerInfo(layer) {
+    currentLayer = layer;
+    var title = /** @type {string} */ (layer.get('label'));
+    var localMetadata = /** @type {Object.<string>} */
+        (layer.get('metadata'));
+    var metadataUid = /** @type {string} */
+        (localMetadata['metadata_id']);
+    popup.setTitle(title);
+    var currentLanguage = /** @type {string} */
+        (gettextCatalog.currentLanguage);
+    var promiseKey = /** @type {string} */
+        (metadataUid + '##' + currentLanguage);
+
+    if (!(promiseKey in promises_)) {
+      promises_[promiseKey] = $http.jsonp(
+          'http://shop.geoportail.lu/Portail/inspire/webservices/getMD.jsp',
+          {params: {
+            'uid': metadataUid,
+            'lang': currentLanguage,
+            'cb': 'JSON_CALLBACK'
+          }}).then(
+              goog.bind(function(resp) {
+
+                var content = {
+                  'uid' : localMetadata['metadata_id'],
+                  'legendUrl' : null,
+                  'hasLegend' : false,
+                  'isError' : false,
+                  'isShortDesc' : true,
+                  'layerMetadata' : null
+                };
+
+                var remoteMetadata = resp.data.root[0];
+                content['layerMetadata'] = remoteMetadata;
+
+                if ('legend_name' in localMetadata) {
+                  content['legendUrl'] = $sce.trustAsResourceUrl(
+                      '//wiki.geoportail.lu/doku.php?id=' +
+                      gettextCatalog.currentLanguage + ':legend:' +
+                      localMetadata['legend_name'] + '&do=export_html'
+                  );
+                  content['hasLegend'] = true;
+                }
+
+                showPopup(content);
```

Comment:
You don't need to call `showPopup` here, as it will be called later when the promise is resolved. I think `showPopup` is called twice currently.